src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -266,6 +267,31 @@ class PythonSetup(Subsystem):
             """
         ),
         advanced=True,
+        removal_version="2.15.0.dev0",
+        removal_hint=softwrap(
+            f"""
+            Pants will soon only support generating lockfiles via the Pex format, as
+            Poetry-generated lockfiles mismatch with Pants's pip-based approach.
+
+            If you do not want to use Pex lockfiles, you will still be able to manually generate
+            lockfiles, e.g. by manually running `poetry export --dev` on your `poetry.lock`. See
+            {doc_url("python-third-party-dependencies#manually-generating-lockfiles")} for more
+            information.
+
+            While Pex generates locks in a proprietary JSON format, you can use the
```

Comment:
How is this relevant? The relevant bit is that _Pants_ uses a proprietary header. If it didn;t you could use `pex ...` to interact with the lock file directly. This is the same as if Pants used proprietary Poetry lockfiles, etc.

Reply:
I was thinking about @asherf's experience migrating from Poetry to Pex. His main issue was how to integrate w/ pip-audit, and he had no idea how to approach it. So I want to mention integrations like this.

I could add a sentence mentioning this?

> Alternatively, you can temporarily remove Pants's comment header and run `pex3 lock export path/to/lock_file.lock`.

Reply:
I have no string opinion on what you write, I just am fairly insistent its true / relevant and the bit you have was not.